indexing-hadoop/src/main/java/io/druid/indexer/InputRowHelper.java

```diff
@@ -0,0 +1,259 @@
+import io.druid.data.input.Row;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.segment.incremental.IncrementalIndex;
+import io.druid.segment.serde.ComplexMetricSerde;
+import io.druid.segment.serde.ComplexMetrics;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.MapWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.UTF8;
+import org.apache.hadoop.io.Writable;
+import org.joda.time.DateTime;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
+public class InputRowHelper
+{
+  private static final Logger log = new Logger(InputRowHelper.class);
+
+  private static final String[] EMPTY_STR_ARRAY = new String[0];
+
+  public final static byte[] toBytes(InputRow row, AggregatorFactory[] aggs)
+  {
+    try {
+      ByteArrayDataOutput out = ByteStreams.newDataOutput();
+
+      List<String> dimList = row.getDimensions();
+      String[] dims = dimList == null ? EMPTY_STR_ARRAY : dimList.toArray(EMPTY_STR_ARRAY);
+      StringArrayWritable sw = new StringArrayWritable(dims);
+      sw.write(out);
+
+      out.writeLong(row.getTimestampFromEpoch());
+
+      MapWritable mw = new MapWritable();
+
+      //writing all dimensions
+      for (String dim : dims) {
+        List<String> dimValue = row.getDimension(dim);
+
+        if (dimValue == null || dimValue.size() == 0) {
+          continue;
+        }
+
+        if (dimValue.size() == 1) {
+          mw.put(new Text(dim), new Text(dimValue.get(0)));
+        } else {
+          mw.put(new Text(dim), new StringArrayWritable(dimValue.toArray(EMPTY_STR_ARRAY)));
+        }
+      }
+
+      //writing all metrics
+      for (AggregatorFactory aggFactory : aggs) {
+        String k = aggFactory.getName();
+
+        ThreadLocal<InputRow> trow = new ThreadLocal<>();
+        trow.set(row);
+        Aggregator agg = aggFactory.factorize(
+            IncrementalIndex.makeColumnSelectorFactory(
+                aggFactory,
+                trow,
+                true
+            )
+        );
+        agg.aggregate();
+
+        String t = aggFactory.getTypeName();
+
+        if (t.equals("float")) {
+          mw.put(new Text(k), new FloatWritable(agg.getFloat()));
+        } else if (t.equals("long")) {
+          mw.put(new Text(k), new LongWritable(agg.getLong()));
+        } else {
+          //its a complex metric
+          Object sketch = agg.get();
+          ComplexMetricSerde serde = ComplexMetrics.getSerdeForType(t);
+          if (serde == null) {
+            throw new ISE("Unknown type[%s]", t);
+          }
+          mw.put(new Text(k), new BytesWritable(serde.getObjectStrategy().toBytes(sketch)));
```

Comment:
@drcrallen spoke to @cheddar and it seems getObjectStrategy() was deprecated to remove the perception that complex aggregator is always persisted using GenericIndexed. So, to still have the functionality of sketch serde, we decided to add toBytes(..) and fromBytes(..) to ComplexMetricSerde and use those instead of getObjectStrategy().
In short, usage of deprecated getObjectStrategy() is removed.